### PR TITLE
ENH migrate to gfortran 7 for osx

### PR DESCRIPTION
**Pull request by beckermr**
This PR has code to migrate to gfortran 7 for osx. I've made an associated PR on the global pinning files [here](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/252).

pinging 

@scopatz 
@CJ-Wright 
@isuruf 

for viz and guidance

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Has there been a change in the pinnings for this?

**Comment by scopatz** [comment]
Can this go in before the pinnings?

**Comment by isuruf** [comment]
Not unless CFEP 9 goes through

**Comment by beckermr** [comment]
I am confused by this convo @scopatz and @isuruf. I am aware of the CFEP 9 proposal, but this migration is separate from that more general solution. Did I miss something?

**Comment by beckermr** [comment]
Ohhhhhhhhh. NM. Asking if this PR can go in before the pinnings PR. Yeah it cannot.

**Comment by beckermr** [comment]
It appears the `r-base` migration is done: https://github.com/regro/cf-scripts/pull/561. 

Shall we merge this and start the gfortran 7 osx one?

**Comment by beckermr** [comment]
@isuruf @scopatz for viz

**Comment by isuruf** [comment]
Waiting on https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/252

**Comment by beckermr** [comment]
Ahhh yes. That has to go first. I just resolved the conflict on that PR. It is ready to merge as far as I am concerned. I will bump people there.

File: conda_forge_tick/auto_tick.xsh [hunk 1]
```diff
@@ -446,6 +446,7 @@ def initialize_migrators(do_rebuild=False):
 
     add_arch_migrate($MIGRATORS, gx)
     add_rebuild_openssl($MIGRATORS, gx)
+    add_rebuild_successors($MIGRATORS, gx, 'fortran_compiler_stub', '7')
```
Comment (isuruf) [inline conda_forge_tick/auto_tick.xsh]: This would make a confusing PR title with `fortran_compiler_stub`.

Comment (beckermr) [reply conda_forge_tick/auto_tick.xsh]: Doesn't this string have to match how the bot mangles the compiler versions internally when figuring out how to migrate things?

Comment (isuruf) [reply conda_forge_tick/auto_tick.xsh]: Yes, you'll need to subclass the rebuild class and change the PR title.

Comment (beckermr) [reply conda_forge_tick/auto_tick.xsh]: Ahhhh ok. I understand the code correctly. Yay!

**Comment by beckermr** [comment]
Merge conflict fixed!

**Comment by beckermr** [comment]
Alright @ocefpaf. I think we are ready to merge this one too!

**Pull request merged**